src/server/scripts/EasternKingdoms/SunwellPlateau/boss_muru.cpp

```diff
@@ -110,53 +139,73 @@ class boss_entropius : public CreatureScript
 
         void Reset() override
         {
-            DoCastAOE(SPELL_NEGATIVE_ENERGY_E);
+            _Reset();
+            DoCast(me, SPELL_ENTROPIUS_COSMETIC_SPAWN, true);
         }
 
-        void EnterCombat(Unit* /*who*/) override
+        void ScheduleTasks() override
         {
-            _EnterCombat();
-            DoCastAOE(SPELL_NEGATIVE_ENERGY_E, true);
-            DoCast(me, SPELL_ENTROPIUS_SPAWN);
-            events.ScheduleEvent(EVENT_SUMMON_BLACK_HOLE, 15000);
+            scheduler.Schedule(Milliseconds(2000), [this](TaskContext /*context*/)
+            {
+                std::list<Creature*> portals;
+                me->GetCreatureListWithEntryInGrid(portals, NPC_MURU_PORTAL_TARGET, 100.0f);
+                for (std::list<Creature*>::iterator itr = portals.begin(); itr != portals.end(); ++itr)
+                    (*itr)->RemoveAllAuras();
+
+                DoCastAOE(SPELL_NEGATIVE_ENERGY_PERIODIC_E, true);
+            });
+
+            scheduler.Schedule(Seconds(15), [this](TaskContext context)
+            {
+                DoCastAOE(SPELL_DARKNESS_E, true);
+                DoCastAOE(SPELL_BLACKHOLE, true);
+
+                context.Repeat();
+            });
         }
 
-        void JustSummoned(Creature* summoned) override
+        void JustSummoned(Creature* summon) override
         {
-            switch (summoned->GetEntry())
+            switch (summon->GetEntry())
             {
                 case NPC_DARK_FIENDS:
-                    summoned->CastSpell(summoned, SPELL_DARKFIEND_VISUAL);
+                    summon->CastSpell(summon, SPELL_DARKFIEND_VISUAL);
                     break;
                 case NPC_DARKNESS:
-                    summoned->AddUnitState(UNIT_STATE_STUNNED);
-                    float x, y, z, o;
-                    summoned->GetHomePosition(x, y, z, o);
-                    me->SummonCreature(NPC_DARK_FIENDS, x, y, z, o, TEMPSUMMON_CORPSE_DESPAWN, 0);
+                    summon->CastSpell(summon, SPELL_BLACKHOLE);
+                    summon->AddUnitState(UNIT_STATE_STUNNED);
```

Comment:
This will be a hack in 99% of cases. `_STUNNED` should usually be applied by core as a spell effect.